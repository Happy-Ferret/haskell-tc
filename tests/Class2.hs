module Class2 where

data String
data Bool = True | False

class Show a where
  show :: a -> String

instance Show Bool

false x = show False

falseFn x = show (x False)

show' x = show x
