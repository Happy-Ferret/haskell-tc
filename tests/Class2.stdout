Bindings:
:: âˆ€ v13. v13 â†’  String
  [4mfalse x = show False[0m
:: v13
  false [4mx[0m = show False
:: âˆ€ v8. .Show v8 â‡’ (Bool â†’  v8) â†’  String
  [4mfalseFn x = show (x False)[0m
:: Bool â†’  v8
  falseFn [4mx[0m = show (x False)
:: âˆ€ v3. .Show v3 â‡’ v3 â†’  String
  [4mshow' x = show x[0m
:: v3
  show' [4mx[0m = show x

Proofs:
coercion: (âˆ€ a. .Show a â‡’ a â†’  String) @ Bool
  false x = [4mshow[0m False
coercion: (âˆ€ a. .Show a â‡’ a â†’  String) @ v8
  falseFn x = [4mshow[0m (x False)
coercion: (âˆ€ a. .Show a â‡’ a â†’  String) @ v3
  show' x = [4mshow[0m x
